File: 6- Maximum subarray/Python/solution3.py
# By using Kadane's algorithm:

# Time complexity: O(n)
# Space complexity: O(1)

def maximumSubarray(arr):
  globalSum = float("-inf")
  localSum = 0
  for element in arr:
    localSum = max(element, localSum + element)
    globalSum = max(globalSum, localSum)
  return globalSum

File: 6- Maximum subarray/Java/solution3.java
// By using Kadane's algorithm:

// Time complexity: O(n)
// Space complexity: O(1)

static int maximumSubarray(int[] arr){
  int globalSum = Integer.MIN_VALUE;
  int localSum = 0;
  for(int element : arr){;
    localSum = Math.max(element, localSum + element);
    globalSum = Math.max(globalSum, localSum);
  }
  return globalSum;
}


